src/NHibernate/Impl/SessionFactoryImpl.cs

```diff
@@ -1276,7 +1276,18 @@ private ICurrentSessionContext BuildCurrentSessionContext()
 				case "web":
 					return new WebSessionContext(this);
 				case "wcf_operation":
+#if NETFX
 					return new WcfOperationSessionContext(this);
+#else
+					// There is no support of WCF Server under .Net Core, so it makes little sense to provide
+					// a WCF OperationContext for it. Since it adds additional heavy dependencies, it has been
+					// considered not desirable to provide it for .Net Standard. (It could be useful in case some
+					// WCF server becames available in another frameworks or if a .Net Framework application
+					// consumes the .Net standard distribution of NHibernate instead of the .Net Framework one)
+					// See https://github.com/dotnet/wcf/issues/1200 and #1842
+					throw new NotSupportedException(
+						"WcfOperationSessionContext is currently supported only by the .Net Framework distribution of NHibernate");
```

Comment:
This `#if #else` added here is currently not strictly needed, since the `WcfOperationSessionContext` constructor under .Net Core or .Net Standard will do the same throw. But since the class is obsoleted under those targets, it avoids referencing an obsolete class.